### PR TITLE
Clean up code for FrequentObjectsStats

### PR DESCRIPTION
**Pull request by LinHu2016**
replace U_32 with uint32_t
 replace UDATA with uintptr_t
 correct code formats

fix:https://github.com/eclipse-openj9/openj9/issues/20449

### CODE REVIEW

**Comment by LinHu2016** [comment]
@dmitripivkine please review the changes, Thanks

File: runtime/gc_stats/FrequentObjectsStats.hpp [hunk 1]
```diff
@@ -92,15 +92,15 @@ class MM_FrequentObjectsStats : public MM_Base
 	 * @param portLibrary the port library
 	 * @param k the number of frequent objects we'd like to accurately report
 	 */
-	MM_FrequentObjectsStats(J9PortLibrary *portLibrary, U_32 k=TOPK_FREQUENT_DEFAULT)
+	MM_FrequentObjectsStats(J9PortLibrary *portLibrary, uint32_t k=TOPK_FREQUENT_DEFAULT)
```
Comment (amicic) [inline runtime/gc_stats/FrequentObjectsStats.hpp:95]: spaces around =

File: runtime/gc_stats/FrequentObjectsStats.cpp [hunk 2]
```diff
@@ -106,24 +106,23 @@ MM_FrequentObjectsStats::traceStats(MM_EnvironmentBase *env)
 
 			utf = J9ROMCLASS_CLASSNAME(arrayClass->leafComponentType->romClass);
 			Trc_MM_FrequentObjectStats_AllocationCacheIndexableObjectAllocation(
-					vmThread, clazz, J9UTF8_LENGTH(utf), J9UTF8_DATA(utf), arity*2, brackets, count, (UDATA) (((float)count)*sampleFreq));
-		}else{
+					vmThread, clazz, J9UTF8_LENGTH(utf), J9UTF8_DATA(utf), arity*2, brackets, count, (uintptr_t) (((float)count)*sampleFreq));
+		} else {
 			Trc_MM_FrequentObjectStats_AllocationCacheObjectAllocation(
-					vmThread, clazz, J9UTF8_LENGTH(J9ROMCLASS_CLASSNAME(clazz->romClass)), J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass)), clazz->totalInstanceSize, count, (UDATA) (((float)count)*sampleFreq));
+					vmThread, clazz, J9UTF8_LENGTH(J9ROMCLASS_CLASSNAME(clazz->romClass)), J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass)), clazz->totalInstanceSize, count, (uintptr_t) (((float)count)*sampleFreq));
```
Comment (amicic) [inline runtime/gc_stats/FrequentObjectsStats.cpp:112]: spaces around operator *

File: runtime/gc_stats/FrequentObjectsStats.cpp [hunk 3]
```diff
@@ -81,23 +81,23 @@ MM_FrequentObjectsStats::kill(MM_EnvironmentBase *env)
 void
 MM_FrequentObjectsStats::traceStats(MM_EnvironmentBase *env)
 {
-	UDATA i;
+	uintptr_t i;
 	J9VMThread *vmThread = (J9VMThread *)env->getOmrVMThread()->_language_vmthread;
 	MM_GCExtensionsBase *extensions = env->getExtensions();
-	/* Should convert easily as frequentObjectAllocationSamplingRate is a UDATA representing a percentage*/
+	/* Should convert easily as frequentObjectAllocationSamplingRate is a uintptr_t representing a percentage*/
 	float sampleFreq = 100/((float) extensions->frequentObjectAllocationSamplingRate);
```
Comment (amicic) [inline runtime/gc_stats/FrequentObjectsStats.cpp:88]: spaces around operator /

**Comment by dmitripivkine** [comment]
Jenkins compile win jdk11

**Review by dmitripivkine — APPROVED** [review]

**Pull request merged**